azurerm/resource_arm_public_ip_prefix.go

```diff
@@ -0,0 +1,191 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-08-01/network"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmPublicIpPrefix() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmPublicIpPrefixCreateUpdate,
+		Read:   resourceArmPublicIpPrefixRead,
+		Update: resourceArmPublicIpPrefixCreateUpdate,
+		Delete: resourceArmPublicIpPrefixDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				id, err := parseAzureResourceID(d.Id())
+				if err != nil {
+					return nil, err
+				}
+				name := id.Path["publicIPPrefixes"]
+				if name == "" {
+					return nil, fmt.Errorf("Error parsing supplied resource id. Please check it and rerun:\n %s", d.Id())
+				}
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"sku": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				Default:          string(network.Standard),
+				DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
We are moving to case sensitivity for properties like this, as such could we more this line?